funnel/models/post.py

```diff
@@ -182,7 +173,7 @@ def __repr__(self):
         label=('unpublished', __("Unpublished")),
     )
 
-    @with_roles(call={'author', 'profile_admin', 'project_editor'})
+    @with_roles(call={'creator', 'profile_admin', 'project_editor'})
```

Comment:
All of these transitions are for the project's editor, not creator. Also, at risk of complicating this model further, we need an additional state: SUBMITTED.

Creator can transition DRAFT to SUBMITTED. Editor can transition DRAFT or SUBMITTED to PUBLISHED (plus the other states).

Reply:
Use `editor` as role here, and map it 1:1 from project if parent is project. Map from `admin` in profile.

Reply:
I think there is a `PostMembership` model coming up as well, to specify author role for users.